requirements.txt

```diff
@@ -9,6 +9,7 @@ dnspython==1.15.0
 docutils==0.14
 dyn==1.8.0
 futures==3.1.1
+google-cloud==0.27.0
```

Comment:
Wow. This pulls in a lot of other dependencies:

> Successfully installed cachetools-2.0.1 certifi-2017.7.27.1 chardet-3.0.4 dill-0.2.7.1 enum34-1.1.6 future-0.16.0 futures-3.1.1 gapic-google-cloud-datastore-v1-0.15.3 gapic-google-cloud-error-reporting-v1beta1-0.15.3 gapic-google-cloud-logging-v2-0.91.3 gapic-google-cloud-pubsub-v1-0.15.4 gapic-google-cloud-spanner-admin-database-v1-0.15.3 gapic-google-cloud-spanner-admin-instance-v1-0.15.3 gapic-google-cloud-spanner-v1-0.15.3 google-auth-1.1.1 google-cloud-0.27.0 google-cloud-bigquery-0.26.0 google-cloud-bigtable-0.26.0 google-cloud-core-0.26.0 google-cloud-datastore-1.2.0 google-cloud-dns-0.26.0 google-cloud-error-reporting-0.26.0 google-cloud-language-0.27.0 google-cloud-logging-1.2.0 google-cloud-monitoring-0.26.0 google-cloud-pubsub-0.27.0 google-cloud-resource-manager-0.26.0 google-cloud-runtimeconfig-0.26.0 google-cloud-spanner-0.26.0 google-cloud-speech-0.28.0 google-cloud-storage-1.3.2 google-cloud-translate-1.1.0 google-cloud-videointelligence-0.25.0 google-cloud-vision-0.26.0 google-gax-0.15.15 google-resumable-media-0.2.3 googleapis-common-protos-1.5.3 grpc-google-iam-v1-0.11.4 grpcio-1.6.3 httplib2-0.10.3 idna-2.6 monotonic-1.3 oauth2client-3.0.0 ply-3.8 proto-google-cloud-datastore-v1-0.90.4 proto-google-cloud-error-reporting-v1beta1-0.15.3 proto-google-cloud-logging-v2-0.91.3 proto-google-cloud-pubsub-v1-0.15.4 proto-google-cloud-spanner-admin-database-v1-0.15.3 proto-google-cloud-spanner-admin-instance-v1-0.15.3 proto-google-cloud-spanner-v1-0.15.3 protobuf-3.4.0 pyasn1-0.3.7 pyasn1-modules-0.1.4 requests-2.18.4 rsa-3.4.2 six-1.11.0 tenacity-4.4.0 urllib3-1.22

Reply:
yes option 2 is using the REST API of google cloud DNS, it's well-documented, and they have got good documentation about various record types including gotchas and the like (https://cloud.google.com/dns/records/json-record) but I'd rather save that for a future version of this provider.

Reply:
No worries, just surprised me when I installed it. Shouldn't hurt anything.